src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private static OperationStrategy operationStrategy;

```

Comment:
The `operationStrategy` should not be static. Since you're passing it through the constructor, you're implying that each `ShopServiceImpl` instance could potentially have a different `OperationStrategy` implementation. Making it static contradicts this and could cause unexpected behavior when multiple instances of `ShopServiceImpl` are used.